modules/authentication/client-react/access/index.js

```diff
@@ -6,19 +6,21 @@ import session from './session';
 
 import AccessModule from './AccessModule';
 
+import resources from './locales';
+
 const ref = React.createRef();
 
-const clearApolloStoreAndReloadComponent = async client => {
-  await client.clearStore();
-  ref.current.reloadPage();
+const resetApolloCache = async client => {
+  await client.cache.reset();
```

Comment:
Why did we change approach of resetting the cache here? We were doing `client.clearStore()`, why did we change it?

Reply:
When a user performs an authorization on the mobile app via social network, the user is being redirected to a third-party authentication service. After the auth completed the user is taken back to the app, but the application, by some reason, doesn't trigger re-update after the cache is cleaned using `client.clearStore()`. Replacing the `client.clearStore()` with `client.cache.reset()` does the trick. I can assume that `clearStore()` functions doesn't do well with subscriptions so that components are not aware about clearing the cahe.

Reply:
"by some reason" - is not a valid explanations. if you do not understand exactly what the problem is you cannot solve it. And your current solution is invalid, because you don't understand what is the exact problem

Reply:
In order to figure out why the `client.clearStore()` behaves different I have made a very deep research and step-by-step logging inside `apollo-client` library. Basically, the `client.clearStore()` invokes `cache.reset()` function under the hood and it should work the same way, but it doesn't. I made assumptions that the reason could in queries and migrations rejections that `client.clearStore()` function makes before `cache.reset()` invocation, so I disabled that functionality but still with no luck - the behavior left the same. Both the `client.clearStore()` and `cache.reset()` invoke the same function of the same object `netCache.reset()` of ApolloCacheRouter as a result. After the `client.clearStore()` being invoked components not re-render. I have inspected every line of cache clearing functionality and so far I don't have any idea what exactly can block subscription invocation for components.

Also, I tried to invoke `client.clearStore()` and `cache.reset()` one be one (in different order) and got the same result as I run `client.clearStore()` only.

Reply:
The comment is incomplete. It should tell that we are calling this method from inside JWTLink in the middle of executing queries and we cannot cancel in-flight queries because of that

Reply:
Done